core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -2267,4 +2267,13 @@ package object config {
       .version("3.3.0")
       .intConf
       .createWithDefault(5)
+
+  private[spark]val MAX_RDD_NAME_LENGTH = ConfigBuilder("spark.rdd.nameMaxLength")
+    .internal()
+    .doc("Maximum number of characters for RDD name. For example, some of the HadoopRDD API will" +
+      "use the path parameter as RDD name which could be extremely long")
+    .version("3.3.0")
+    .intConf
+    .checkValue(_ > 3, "This value must be bigger than 3.")
+    .createWithDefault(256)
```

Comment:
@yaooqinn .
- This looks like a breaking change for the downstream expecting the same names they set.
- If we need this configuration, the default value should be `Int.MaxValue` at least too.

Reply:
yes, I had intended to change the spark auto-generated ones, such as

https://github.com/apache/spark/blob/abecdfe831805fb53fa949558ea7a5ca9042e465/core/src/main/scala/org/apache/spark/SparkContext.scala#L992

https://github.com/apache/spark/blob/abecdfe831805fb53fa949558ea7a5ca9042e465/core/src/main/scala/org/apache/spark/SparkContext.scala#L1044

https://github.com/apache/spark/blob/abecdfe831805fb53fa949558ea7a5ca9042e465/core/src/main/scala/org/apache/spark/SparkContext.scala#L1269

 How about that?

Reply:
It might be good to make them shorter and more meaningful, but is this not ultimately a display issue?

Reply:
> but is this not ultimately a display issue?

Yes, let me try to fix it at UI side